File: src/unity/Runtime/FirebasePerformance/Internal/FirebasePerformanceTrace.cs
namespace EE.Internal {
    internal class FirebasePerformanceTrace : IFirebasePerformanceTrace {
        private readonly IMessageBridge _bridge;
        private readonly FirebasePerformance _plugin;
        private readonly string _name;

        public FirebasePerformanceTrace(
            IMessageBridge bridge,
            FirebasePerformance plugin,
            string name) {
            _bridge = bridge;
            _plugin = plugin;
            _name = name;
        }

        public void Start() {
            throw new System.NotImplementedException();
        }

        public void Stop() {
            throw new System.NotImplementedException();
        }

        public void PutMetric(string metricName, long value) {
            throw new System.NotImplementedException();
        }

        public void IncrementMetric(string metricName, long incrementBy) {
            throw new System.NotImplementedException();
        }

        public long GetLongMetric(string metricName) {
            throw new System.NotImplementedException();
        }
    }
}

File: src/ts/src/firebase_performance/internal/FirebasePerformanceTrace.ts
import { IMessageBridge } from "../../core";
import { IFirebasePerformanceTrace } from "../IFirebasePerformanceTrace";
import { FirebasePerformance } from "./FirebasePerformance";

export class FirebasePerformanceTrace implements IFirebasePerformanceTrace {
    private readonly _bridge: IMessageBridge;
    private readonly _plugin: FirebasePerformance;
    private readonly _name: string;

    public constructor(bridge: IMessageBridge, plugin: FirebasePerformance, name: string) {
        this._bridge = bridge;
        this._plugin = plugin;
        this._name = name;
    }

    public start(): void {
        this._bridge.call(`FirebasePerformance_start_${this._name}`);
    }

    public stop(): void {
        this._bridge.call(`FirebasePerformance_stop_${this._name}`);
    }

    public putMetric(metricName: string, value: number): void {
        const request = {
            key: metricName,
            value: value,
        };
        this._bridge.call(`FirebasePerformance_putMetric_${this._name}`, JSON.stringify(request));
    }

    public incrementMetric(metricName: string, incrementBy: number): void {
        const request = {
            key: metricName,
            value: incrementBy,
        };
        this._bridge.call(`FirebasePerformance_incrementMetric_${this._name}`, JSON.stringify(request));
    }

    public getLongMetric(metricName: string): number {
        const response = this._bridge.call(`FirebasePerformance_getLongMetric_${this._name}`, metricName);
        return parseInt(response);
    }
}